compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -3295,43 +3298,52 @@ public void visit(BLangCheckedExpr checkedExpr, AnalyzerData data) {
         data.failVisited = true;
         analyzeExpr(checkedExpr.expr, data);
 
+        DefaultValueState defaultValueState = data.defaultValueState;
         if (data.env.scope.owner.getKind() == SymbolKind.PACKAGE) {
+            if (defaultValueState == DefaultValueState.PARAMETER_DEFAULT) {
+                dlog.error(checkedExpr.pos, DiagnosticErrorCode.INVALID_USAGE_OF_CHECK_IN_PARAMETER_DEFAULT);
+            }
             // Check at module level.
             return;
         }
```

Comment:
Would it make sense to update the code like this and also remove lines 3310-3312. Other than that LGTM
```suggestion
        if (defaultValueState == DefaultValueState.PARAMETER_DEFAULT) {
            dlog.error(checkedExpr.pos, DiagnosticErrorCode.INVALID_USAGE_OF_CHECK_IN_PARAMETER_DEFAULT);
            return;
        }

        if (data.env.scope.owner.getKind() == SymbolKind.PACKAGE) {
            return;
        }
```